src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -103,6 +103,11 @@ public function request(string $method, string $url, array $options = []): Respo
         [$url, $options] = self::prepareRequest($method, $url, $options, $this->defaultOptions);
         $scheme = $url['scheme'];
         $authority = $url['authority'];
+
+        if (null === $authority) {
```

Comment:
could this check be shared with all implementations somehow? something in prepareRequest for example? I don't remember the details of this method :)

Reply:
makes sense indeed, moving the test without applying the patch there led to errors in most of the other implementations